### DISCUSSION TITLE
Can we access to related fields in @Root ?

### DISCUSSION THREAD

**Discussion by MaxiSantos** [MichalLytek/typegraphql-prisma#224] [Q&A] [answered]
Hello, I'm trying to filter some nested field but can't figure it out how to do it. 

I saw this post: https://github.com/MichalLytek/typegraphql-prisma/issues/126 and that's exactly what I'm looking for.

This is the query I sent from my frontend
```js
stores(where:{productId:{some:{id:{equals:8}}}}){
    id
    name
    products{
      storeId
      productId
      name
    }
```

I want to get all stores filtered by productId but also I want products to only match the given product.

So I tried to filter it with prisma in my backend

```js
@Query((returns) => UserSearchResponse, { nullable: true })
  async search2(
    @Ctx() { prisma }: typeof context,
    @Args() { userId, productId, categoryId, query, location }: SearchArgs,    
  ): Promise<UserSearchResponse | null> {

...
const stores = await prisma.store.findMany({
        where: {        
          product: {
            some: {
              status: true,
              productId,
            },
          },
        },
        include: {
          product: {where: {id: productId}}
        }
      })
...
}
```

but this isn't working. Then I tried to create a custom field resolver

```js
@Resolver(of => Store)
export class CustomStoreResolver {
  // queries and mutations

  @FieldResolver(type => Product, { nullable: true })
  async productFiltered(
    @Ctx() { prisma }: typeof context,
    @Arg("productId", type => Int) productId: number,
    @Root() store: Store
  ) {
    const [filteredProduct] = await prisma.store
      .findUnique({ where: { id: store.id } })
      .product({ where: { productId } });

    //return store.product.filter(item => item.productId === productId);
    return filteredProduct;
  }
}
```

I would have expected to get access to related fields on the Root object but I only have access to scalars field. So how can I get access to product so I avoid having to make an extra prisma call ?

This is the updated query sent from the frontend

```js
stores(where:{productId:{some:{id:{equals:8}}}}){
    id
    name
    productFiltered(productId: 8){
      storeId
      productId
      name
    }
```

**Comment by MaxiSantos** [comment]
@MichalLytek based on this comment I'm assuming creating a custom field resolver is the way to go https://github.com/MichalLytek/typegraphql-prisma/discussions/112#discussioncomment-722537 since it looks I can't filter on relational fields, right? Is there a way to get access to a relational field from the Root object?

**Comment by MichalLytek** [comment]
Relation resolver has support for filtering arguments:
https://github.com/MichalLytek/typegraphql-prisma/blob/main/experiments/prisma/generated/type-graphql/resolvers/relations/MainUser/MainUserRelationsResolver.ts
https://github.com/MichalLytek/typegraphql-prisma/blob/main/experiments/prisma/generated/type-graphql/resolvers/relations/MainUser/args/MainUserPostsArgs.ts

**Reply by MaxiSantos** [reply]
Thanks for the quick response. Yeah I saw that but I see in the example you are using

```js
getPrismaFromContext(ctx).user.findUnique
```
In my case I'm using

```js
prisma.store.findMany
``` 

Is there a limitation with findMany ?

I can confirm the example is working because is what I'm using in my 'CustomStoreResolver'
```js
const [filteredProduct] = await prisma.store
      .findUnique({ where: { id: store.id } })
      .product({ where: { productId } });
```

**Reply by MaxiSantos** [reply]
here's another comment related that mention I can't have access to related fields https://github.com/MichalLytek/typegraphql-prisma/discussions/104#discussioncomment-636473 and I should use field resolvers as you suggested in previous comment. But it seems there is a limitation with `findMany`

**Comment by MaxiSantos** [comment, marked as answer]
Well.. don't know what to say. Now I can have access to relational fields in the @Root object. Not sure why I counld't before. The only change I suspect is related is this

I use to use this

```js
import {
  crudResolvers, relationResolvers, ResolversEnhanceMap,
  applyResolversEnhanceMap
} from '../generated/type-graphql';

const schema = await tq.buildSchema({
    resolvers: [
      ...crudResolvers,
      ...relationResolvers,
     other resolvers,
    ],
    authChecker: customAuthChecker,
    authMode: "error",
    validate: true,
    globalMiddlewares: [ErrorInterceptor],
  });
```

and now I have this
```js
import {
  resolvers, ResolversEnhanceMap,
  applyResolversEnhanceMap
} from '../generated/type-graphql';

const schema = await tq.buildSchema({
    resolvers: [
      ...resolvers,
      other resovers
    ],
    authChecker: customAuthChecker,
    authMode: "error",
    validate: true,
    globalMiddlewares: [ErrorInterceptor],
  });
```

I'd like to avoid having to create a custom field resolver but for now I only was cared about not having to make that extra prisma query. I'd love to hear if it's possible to use filters on fields when you use `findMany` the example you gave me seems to only work with findUnique like in the docs https://prisma.typegraphql.com/docs/advanced/adding-fields
